src/figuresClasses.ts

```diff
@@ -1,19 +1,81 @@
-export interface Figure {
+/* import { publicDecrypt } from "crypto";
+import { type } from "os"; */
 
-}
+type FigureShape = 'triangle' | 'circle' | 'rectangle';
 
-export class Triangle {
+type FigureColor = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: FigureShape;
+  color: FigureColor;
+  getArea(): number;
 }
 
-export class Circle {
+export class Triangle implements Figure {
+  shape:FigureShape = 'triangle';
+
+  constructor(
+    public color: FigureColor,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Side length is not valid');
+    }
+
+    if (a >= b + c || b >= a + c || c >= a + b) {
+      throw new Error('sum of two sides must always be greater than the third');
+    }
```

Comment:
would be nice to have a separate method like `validateTriangle` for that purpose